File: sort/selection/selection-sort.py
def selection_sort(arr) -> list:
    high = -1
    ind_high = -1
    for index in reversed(range(0,len(arr))):
        for s_index in range(0,index+1):
            val = arr[s_index]
            if val > high:
                high = val
                ind_high = s_index
        if ind_high != -1:
            temp = arr[index]
            arr[index] = high
            arr[ind_high] = temp
        high = -1
        ind_high = -1
    return arr

if __name__=="__main__":
    array = []
    while True:
        temp = input("Enter a value into the array (press enter to finish): \n")
        try:
            if temp:
                array.append(int(temp))
            else:
                break
        except:
            print("integers only please")

    print(selection_sort(array) if array else "Please give a valid entry")

File: sort/selection/selectionSort.go
package main

type High struct {
	index int
	value int
}

func selection_sort(arr []int) []int {
	for i := (len(arr) - 1); i >= 0; i-- {
		high := High{index: -1, value: -1}
		for j := 0; j < i; j++ {
			val := arr[j]
			if val > high.value {
				high.value = val
				high.index = j
			}
		}
		if high.index != -1 {
			arr[high.index], arr[i] = arr[i], arr[high.index]
		}
	}
	return arr
}
